app/controllers/api/v1/labels.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'doorkeeper/grape/helpers'
+
+module API
+  module V1
+    class Labels < Grape::API
+      helpers Doorkeeper::Grape::Helpers
+
+      before { doorkeeper_authorize! }
+
+      before do
+        def current_account
+          Account.find(doorkeeper_token.resource_owner_id)
+        end
+      end
+
+      resource :labels do
+        desc 'List all labels for current account.'
+        get do
+          present current_account.labels, with: API::Entities::Label
+        end
+
+        desc 'Return a label by key.'
+        params do
+          requires :key, type: String, allow_blank: false, desc: 'Label key.'
```

Comment:
This is a security hole, aren't you missing current_account
you can only see your labels.

Reply:
@mod, do `restricted` (private) labels always has an account assigned? I wasn't sure so made a general assumption. If `restricted` always has account assigned, then this will be fixed

Reply:
fixed